src/python/BuildingControlsSimulator/DataClients/WeatherChannel.py

```diff
@@ -125,14 +136,34 @@ def make_epw_file(
                 "NREL_EPLUS" + f"_{sim_config['identifier']}" + f"_{fill_epw_fname}",
             )
 
-            # fill any missing fields in epw
+            # add nsrdb solar fields
             # need to pass in original dyd datetime column name
-            epw_data = self.fill_epw(
-                input_epw_data=self.data,
-                datetime_channel=datetime_channel,
-                fill_epw_data=fill_epw_data,
-                sim_config=sim_config,
-            )
+            if not fill_nsrdb_data.empty:
+                nsrdb_data = self.fill_nsrdb(
```

Comment:
We should decouple the nsrdb from epw generation. The call to fill nsrdb could be in DataClient.get_data(), after the initialization of the WeatherChannel instance.

Reply:
can do.

Reply:
done.